### PR TITLE
Implement node event handler

### LINKED ISSUE #276: Add event handler for `Node` Events.
`Node` change events are now available in the event stream of the G-Research fork of yunikorn. Implement event handler for Node Events.

### PR DESCRIPTION
**Pull request by sudiptob2**
closes #276

### CODE REVIEW

File: internal/database/repository/node.go [hunk 1]
```diff
@@ -40,56 +59,231 @@ func applyNodeFilters(builder *sql.Builder, filters NodeFilters) {
...
+			return nil, fmt.Errorf("could not scan node utilizations from DB: %v", err)
+		}
+		nodesUtil = append(nodesUtil, &nu)
+	}
+	return nodesUtil, nil
+}
+
+func (s *PostgresRepository) GetNodesPerPartition(ctx context.Context, partition string, filters NodeFilters) ([]*model.Node, error) {
 	queryBuilder := sql.NewBuilder().
 		SelectAll("nodes", "").
```
Comment (nikola-jokic) [inline internal/database/repository/node.go:281]: Not in scope of this PR, but I'm not a huge fan of scanning based on position. We should probably change the interface to list the fields. We shouldn't look into the db field order to make the scan correct. 

WDYT @richscott @sudiptob2

File: internal/database/repository/node.go [hunk 2]
```diff
@@ -40,56 +59,231 @@ func applyNodeFilters(builder *sql.Builder, filters NodeFilters) {
...
+		q,
+		pgx.NamedArgs{
+			"deleted_at_nano": deletedAtNano,
+			"ids":             ids,
+		},
+	)
+	return err
+}
+
+func (s *PostgresRepository) InsertNodeUtilizations(
```
Comment (nikola-jokic) [inline internal/database/repository/node.go:224]: Do we need insert of node utilizations?

**Review by nikola-jokic — CHANGES_REQUESTED** [review]
Had few questions and suggestions

File: internal/database/repository/node.go [hunk 3]
```diff
@@ -40,56 +59,231 @@ func applyNodeFilters(builder *sql.Builder, filters NodeFilters) {
...
-			return fmt.Errorf("could not insert application into DB: %v", err)
+			return fmt.Errorf("could not insert node utilizations into DB: %v", err)
 		}
+
 	}
 	return nil
 }
 
-func (s *PostgresRepository) GetNodesPerPartition(ctx context.Context, partition string, filters NodeFilters) ([]*dao.NodeDAOInfo, error) {
+func (s *PostgresRepository) GetNodeUtilizations(
```
[the remark this answers is not in the document]
Comment (sudiptob2) [reply internal/database/repository/node.go]: Thanks! messed up this one while rebasing.

File: internal/database/repository/node.go (hunk 1 above)
Comment (sudiptob2) [reply internal/database/repository/node.go:206]: Good Idea, I think depending on the order is error-prone.

File: internal/model/node.go [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+	n.Allocated = nodeInfo.Allocated
+	n.Occupied = nodeInfo.Occupied
+	n.Available = nodeInfo.Available
+	n.Utilized = nodeInfo.Utilized
+	n.Allocations = nodeInfo.Allocations
+	n.Schedulable = nodeInfo.Schedulable
+	n.IsReserved = nodeInfo.IsReserved
+
+	lookup := make(map[string]struct{})
+	if len(nodeInfo.Allocations) > 0 {
```
[the remark this answers is not in the document]
Comment (sudiptob2) [reply internal/model/node.go:27]: Added test

**Review by nikola-jokic — APPROVED** [review]
lgtm!

**Pull request merged**